### PR TITLE
DM-43906: Make gauss2d more dev guide compliant

### PR DESCRIPTION
**Pull request by taranu**
This is in prep for adding it to lsst_distrib. Although it will be added as a third party package to start, the dev guidelines are worth following anyway.

### CODE REVIEW

File: tests/test_to_string.cc [hunk 1]
```diff
@@ -0,0 +1,30 @@
+#define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
```
Comment (arunkannawadi) [inline tests/test_to_string.cc:1]: It's been customary for all the test files to also have the preamble as the main code. I don't see an explicit recommendation on the dev guide about this though.

**Review by arunkannawadi — APPROVED** [review]
Looks mostly good. A few comments/suggestions:

- A single blank line at the end of file seems to be missing in several files. It'd be good to make sure that `flake8` could prevent such files from getting merge to `main`.
- Should the various `.sh` files go into `bin.src/` folder or may be `scripts/` folder?
- Should `python/` contain `lib/include/` path within it?
- Going forth, do you need an `ellipse` module here or would the one from `afw` suffice?
- I have no idea about how/where `meson` files should be placed.

File: include/lsst/gauss2d/type_name.h [hunk 2]
```diff
@@ -21,14 +22,16 @@
...
 
+#include <string>
 #include <string_view>
 
-namespace gauss2d {
+#include "string_utils.h"
+
+namespace lsst::gauss2d {
 
-// https://stackoverflow.com/questions/81870/is-it-possible-to-print-a-variables-type-in-standard-c/64490578#64490578
```
[the remark this answers is not in the document]
Comment (taranu) [reply include/lsst/gauss2d/type_name.h:31]: Hmm, not sure why I removed it - I have the same file in modelfit_parameters (lacking anywhere to put common code between the two). But anyway I replaced it with the new shorter permalink.

File: .gitignore [hunk 3]
```diff
@@ -3,7 +3,12 @@ build
 builddir
 build-debug
 build-release
-python/gauss2d.egg-info/
+python/src/lsst_gauss2d.egg-info/
+
+# SCons
+config.log
+.sconf_temp
+.sconsign.dblite
```
[the remark this answers is not in the document]
Comment (taranu) [reply .gitignore:11]: No, I manually added that file. It is now essentially provided by documenteer. There is an xml entry in the `doc/.gitignore`.

**Comment by taranu** [comment]
Thanks for the review.

>     * A single blank line at the end of file seems to be missing in several files. It'd be good to make sure that `flake8` could prevent such files from getting merge to `main`.

I added a rule to `.clang-format` for that (this seems to be a relatively new LLVM feature). I'm not planning to use flake8 for the build and other files, though.

>     * Should the various `.sh` files go into `bin.src/` folder or may be `scripts/` folder?

They're convenience build scripts, not CLI utilities or anything else that belongs on the path. Having said that there are more of them than I'd like; I would condense them but I'm reluctant to make shell scripts any more complicated than they need to be.

>     * Should `python/` contain `lib/include/` path within it?

Yes, it's a mini-library purely for generating pybind11 bindings for templated classes (e.g. the GSL image classes forthcoming in gauss2d_fit).

>     * Going forth, do you need an `ellipse` module here or would the one from `afw` suffice?

I'm not sure if afw provides all of the functionality I need, and it's mostly a moot point unless afw becomes standalone. Having said that, I could add convenience functions/constructors to convert between the two if needed.

>     * I have no idea about how/where `meson` files should be placed.

`meson.build` is equivalent to a `SConstruct`. The pattern of using separate `meson.build` files in subdirectories and calling `subdir("subdir")` is not strictly necessary but is more or less standard amongst meson users.

**Pull request merged**